internal/resources/webform/webform.js

```diff
@@ -407,11 +407,27 @@ window.initGRPCForm = function(services, invokeURI, metadataURI, debug, headers)
         input.childKeys = [];
 
         var arrayVal = [];
-        for (k in value) {
+        for (var k in value) {
             if (value.hasOwnProperty(k)) {
+                var keyField = mapEntryFields[0];
+                var valField = mapEntryFields[1];
+                if (keyField.type !== "string") {
+                    // k should be a number or a bool
+                    try {
+                        k = JSON.parse(k);
+                        // success!
+                    } catch {
+                        // failure! leave k alone
+                    }
+                }
+                var v = value[k];
```

Comment:
Does this actually work? Converting k to a different type and then doing a lookup in the map that originally had string keys? I guess so in Javascript..

You also could just do `const [k, v] of Object.entries(value)` if you wanted.

Reply:
It does work. But I can see why it's confusing; I can move this line above the conversion of `k`.

Regarding your suggestion, neat, I didn't realize that was a thing -- maybe EcmaScript 6? As you can see, my JS knowledge is for ES 5 (I haven't done much JS since the new stuff was introduced....).